deploy/openshift/deployment.yaml

```diff
@@ -25,6 +25,33 @@ spec:
     spec:
       serviceAccount: kiali
       containers:
+      - image: openshift/oauth-proxy:v1.1.0
+        name: kiali-proxy
+        args:
+        - -provider=openshift
+        - -https-address=:20002
+        - -http-address=
+        - -email-domain=*
+        - -upstream=http://localhost:20001
+        - -openshift-service-account=kiali
+        - '-openshift-sar={"resource": "namespaces", "verb": "get"}'
```

Comment:
I don't think we want this. We don't want the proxy to be doing any sort of permission checks. We just need to the token and then Kiali uses that to do all the calls to the k8s master. That will control who has access to what.

This isn't going to work for our usecase since this means the user who accesses Kiali has to have permission to get all the namespaces (which isn't a normal user)

Reply:
I think we can fix this for 1186, and for this one just have one permission that might be enough for our user. I'd say it's the responsibility of the proxy to do that, else we need to check the permission directly, on every request, where they only check it on the moment of the auth.

Reply:
>I'd say it's the responsibility of the proxy to do that, else we need to check the permission directly, on every request, where they only check it on the moment of the auth.

With KIALI-1186 we don't want to check the permission, we just use the token provided directly when communicating with the k8s master. The k8s master verifies the permissions before returning the results.

I don't know why we want this proxy when we get to KIALI-1186 since we just want to do a normal OAuth workflow where we request the token from OpenShift and use that instead of the Kiali service account token.

Reply:
Yeah, that's the confusing part: the service account is necessary to create the correct oauth configuration. It does not use the service account token, but the user one. Having oauth directly, without using the service account is unstable, in the sense that it doesn't always return to the correct page, and for some reason, the returned token is not an access token that we can use, but a HMAC signed key that we can use for auth, but not for connecting.

Also, this proxy is the same that Prometheus uses on Openshift (https://github.com/openshift/cluster-monitoring-operator/blob/9c97908591f4b4fc3b12251c4b6ab0d6289fdbc0/assets/prometheus-k8s/prometheus.yaml) and the one that Jaeger is evaluating, afaik.

Reply:
Yes, its the same proxy that others are using or considering, but its a different usecase than ours.